src/python/grpcio/grpc/__init__.py

```diff
@@ -13,17 +13,43 @@
 # limitations under the License.
 """gRPC's Python API."""
 
+from __future__ import annotations
+
 import abc
+from concurrent import futures
 import contextlib
 import enum
 import logging
 import sys
+import threading
+import types
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterator,
+    Mapping,
+    NoReturn,
+    Optional,
+    Sequence,
+    Tuple,
+)
 
 from grpc import _compression
 from grpc._cython import cygrpc as _cygrpc
 from grpc._runtime_protos import protos
 from grpc._runtime_protos import protos_and_services
 from grpc._runtime_protos import services
+from grpc._typing import ArityAgnosticMethodHandler
```

Comment:
Since we're making these public for the first time, we should consider whether we want to make them available as (for example) `grpc.ArityAgnosticMethodHandler` (as we're doing here) or `grpc.typing.ArityAgnosticMethodHandler`. I think I'd prefer the submodule for purely organizational reasons. We would then need to figure out whether to introduce the `grpc.typing` module _in addition_ to `grpc._typing` or to rename `grpc._typing` to `grpc.typing` and prepend an underscore to all types that we want to keep internal. I think I'd prefer the renaming option, again, for organizational purposes.